torchao/dtypes/affine_quantized_tensor.py

```diff
@@ -228,6 +228,49 @@ def from_float(
             dtype=input_float.dtype
         )
 
+    @classmethod
+    def from_float_static(
+        cls,
+        input_float: torch.Tensor,
+        scale: torch.Tensor,
+        zero_point: torch.Tensor,
+        block_size: Tuple[int, ...],
+        target_dtype: torch.dtype,
+        quant_min: Optional[int] = None,
+        quant_max: Optional[int]  = None,
+        zero_point_domain: ZeroPointDomain = ZeroPointDomain.INT,
+        extended_layout: str = "plain",
+        # TODO: this is only for "tensor_core_tiled", need to figure out
+        # the proper API for this arg
+        inner_k_tiles: Optional[int] = None,
+    ):
+        original_shape = input_float.shape
+        if extended_layout == "tensor_core_tiled":
+            orig_out_features, orig_in_features = input_float.shape
+            in_features = find_multiple(orig_in_features, 1024)
```

Comment:
Re the 1024 and 8 padding heuristics as well this is what the NVIDIA docs say https://x.com/marksaroufim/status/1621580671776092160

So this heuristic tends to be dtype and device dependent - it's possible 1024 and 8 are fine but that would be right mostly out of luck

Reply:
this should only be used by the tinygemm use case I think, so we should probably add some verifications